### PR TITLE
feat: Add Digamma Function and tests to Jax Frontend

### LINKED ISSUE #26379: digamma

### PR DESCRIPTION
**Pull request by SalmaEl-gohary**
## Related Issue

Close #26379

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by SalmaEl-gohary** [comment]
Hi @AlexandraPosta, Thanks for the feedback. Yes, I have run the tests locally and I will attach them here. 
As I understood from the Deep Dive, The functions in the frontends are supposed to call the API from Ivy and that's what I applied with the addition of the invalid values. Plz correct me if my understanding is wrong.

File: ivy/functional/frontends/jax/lax/operators.py [hunk 1]
```diff
@@ -330,6 +330,16 @@ def cumsum(operand, axis=None, reverse=False):
...
 
 
+@with_unsupported_dtypes(
+    {"0.4.16 and below": ("bfloat16", "float16", "bool", "complex64", "complex128")},
+    "jax",
+)
+@to_ivy_arrays_and_back
+def digamma(x, name=None):
+    digamma_fun = ivy.digamma
+    return ivy.array(digamma_fun(x), dtype=x.dtype)
```
Comment (hmahmood24) [inline ivy/functional/frontends/jax/lax/operators.py:340]: We can simply do `return ivy.astype(ivy.digamma(x), x.dtype)` here

File: ivy_tests/test_ivy/test_frontends/test_jax/test_lax/test_operators.py [hunk 2]
```diff
@@ -1516,6 +1516,33 @@ def test_jax_cumsum(
     )
 
 
+# digamma
+@handle_frontend_test(
+    fn_tree="jax.lax.digamma",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float")),
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_lax/test_operators.py:1522]: We prever using `"valid"` rather than `"float"` here

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @SalmaEl-gohary, just request some small changes on the PR. Thanks :)

**Comment by SalmaEl-gohary** [comment]
I have made the required changes. Are there any other changes?

File: ivy/functional/frontends/jax/lax/operators.py [hunk 3]
```diff
@@ -330,6 +330,16 @@ def cumsum(operand, axis=None, reverse=False):
     return ivy.cumsum(operand, axis=axis, dtype=operand.dtype)
 
 
+@with_unsupported_dtypes(
+    {"0.4.16 and below": ("bfloat16", "float16", "bool", "complex64", "complex128")},
+    "jax",
+)
+@to_ivy_arrays_and_back
+def digamma(x, name=None):
+    ivy.digamma
```
Comment (hmahmood24) [inline ivy/functional/frontends/jax/lax/operators.py:339]: I think you forgot to remove this line

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @SalmaEl-gohary, did you run your tests locally using `pytest`? The tests are failing when tested locally. We'll need to fix the tests before we can merge the PR. Thanks :)
image

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @SalmaEl-gohary, as I mentioned above, you'll have to look into fixing the tests from your PR first before we can merge the PR. 
image

**Comment by hmahmood24** [comment]
Closing the PR since it has been stale for more than 2 weeks.

**Pull request closed without merging**